src/Files/File.php

```diff
@@ -1014,7 +1014,7 @@ protected function addMessage($error, $message, $line, $column, $code, $data, $s
                     continue;
                 }
 
-                foreach ($patterns as $pattern => $type) {
+                foreach (array_keys($patterns) as $pattern) {
```

Comment:
I decided to also fix unused array values, but this is a bit a matter of taste. It is one more function call to array_keys(), but you get the benefit of removing a variable from scope. That improves maintainability as there can be less accidental name clashes when the code is changed later and a variable with the same name is introduced.

PHP as a language does not warn you when variable names overlap, so I think it is best practice to keep the number of variables in scope so small that the code is still readable.

Reply:
@klausi The `array_*()` functions have notorious performance issues and this is likely the "hottest" function (most used function) in PHPCS, so please undo this change.

> PHP as a language does not warn you when variable names overlap, so I think it is best practice to keep the number of variables in scope so small that the code is still readable.

I'd rather have a call to the inexpensive `isset()` straight after such loops than to change the loop signature itself.

Reply:
alright, I will undo the array_keys() changes!

Reply:
Thanks! (and sorry, I just noticed a typo in my previous comment, that should have been `unset()`, not `isset()`).

Reply:
I've opened a task ticket (#521) to see about adding more `unset()`s where appropriate to prevent the problems you hinted at.